test-framework/common/src/main/java/io/quarkus/test/common/PathTestHelper.java

```diff
@@ -99,6 +100,20 @@
                 File.separator + "test-classes",
                 File.separator + "classes");
         //endregion
+
+        String mappings = System.getenv("ADDITIONAL_TEST_TO_MAIN_MAPPINGS");
```

Comment:
@glefloch Based on my analysis, if we want to keep the code impact to a minimum, then I will need to do the following,

* Identify the test sources and serialize it as part of the plugin (via the tooling API)
* Read that information in this class and associate the test-to-main mapping so that `isInTestDir` returns true

If this approach sounds good, then I can serialize it as part of the [QuarkusPluginExtension.beforeTest()](https://github.com/quarkusio/quarkus/blob/master/devtools/gradle/src/main/java/io/quarkus/gradle/QuarkusPluginExtension.java#L49) method.

Apart from the Gradle extension, is there any additional place that needs to be modified?

Reply:
Yes, actually, we are already scanning workspace here: https://github.com/quarkusio/quarkus/blob/77b772c6dc4d44649fa6b03b1b824bac9abfea5d/devtools/gradle/src/main/java/io/quarkus/gradle/builder/QuarkusModelBuilder.java#L126

If you can add those source set here, you will be able to get them through the model used : https://github.com/quarkusio/quarkus/blob/77b772c6dc4d44649fa6b03b1b824bac9abfea5d/test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java#L199